File: frontend/src/types/device.ts
import { DeviceType } from "../shared/deviceTypes";

export type ConnectionType =
  | "Virtual"
  | "MQTT"
  | "Zigbee";

export type Device = {
  id: string;
  name: string;
  type: DeviceType;
  roomId: string;
  stateJson: string;

  connectionType: ConnectionType;

  mqttTopic?: string;
  externalId?: string;
};

File: SmartHome/Models/Device.cs
using SmartHome.Models.Enums;

namespace SmartHome.Models;

public class Device
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = string.Empty;

    public DeviceType Type { get; set; }

    public ConnectionType ConnectionType { get; set; }
        = ConnectionType.Virtual;

    public string? MqttTopic { get; set; }

    public string? ExternalId { get; set; }

    public Guid RoomId { get; set; }

    public Room Room { get; set; } = null!;

    public Guid UserId { get; set; }

    public string StateJson { get; set; } = "{}";
}